posts/2020-07-24-dev-mode-containers-preview.adoc

```diff
@@ -0,0 +1,74 @@
+
+== Starting dev mode with container support
+
+For Maven projects, specify the Liberty Maven Plugin with version `3.3-M2` in your project’s `pom.xml` file.
+[source,xml]
+----
+    <plugin>
+        <groupId>io.openliberty.tools</groupId>
+        <artifactId>liberty-maven-plugin</artifactId>
+        <version>3.3-M2</version>
+    </plugin>
+----
+
+For Gradle projects, specify the Liberty Gradle Plugin with version `3.1-M1` in your `build.gradle` file.
+[source,groovy]
+----
+apply plugin: 'liberty'
+
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath 'io.openliberty.tools:liberty-gradle-plugin:3.1-M1'
+    }
+}
+----
+
+Create a Dockerfile in your project root directory as shown in link:https://github.com/OpenLiberty/ci.docker#building-an-application-image[Building an Application Image], including COPY commands for your application and configuration files.
+
+Start dev mode with container support by running one of following commands in your project:
+
+* Maven: `mvn liberty:devc`
+* Gradle: `gradle libertyDevc`
+
+This compiles your application, builds the development image, and runs the server in the container.  You can continue to make changes to your source code or configuration files while dev mode is running.  Press Enter in the terminal to run tests, or attach a debugger to port `7777` at any time to debug your application.  Quit dev mode by pressing CTRL+C in the terminal, or type `q` and press Enter.
+
+== Additional resources
+
+For a more comprehensive demo, see the link:https://github.com/OpenLiberty/demo-devmode/tree/devc[devc branch of the demo-devmode project].
+
+For more details on dev mode with container support, see the documentation for the link:https://github.com/OpenLiberty/ci.maven/blob/master/docs/dev.md#devc-container-mode[devc goal of the Liberty Maven Plugin] or the link:https://github.com/OpenLiberty/ci.gradle/blob/master/docs/libertyDev.md#libertydevc-task-container-mode[libertyDevc task of the Liberty Gradle Plugin].
+
+== Next steps
+
+As this is a tech preview, functionality can change and we are open to suggestions.  Give it a try and let us know what works or doesn't work for you.  You can post in the link:https://gitter.im/OpenLiberty/developer-experience[Open Liberty developer experience chat] or open issues in the link:https://github.com/OpenLiberty/ci.maven[Liberty Maven Plugin] or link:https://github.com/OpenLiberty/ci.gradle[Liberty Gradle Plugin] repositories.
```

Comment:
Should we indicate that we look to iteratively provide additional previews for feedback?

Reply:
Added